File: smarttrade_app/lib/features/boleta_inteligente/domain/repositories/instrument_repository.dart
import 'package:dartz/dartz.dart';
import 'package:smarttrade_app/core/error/failures.dart';
import 'package:smarttrade_app/features/boleta_inteligente/domain/entities/instrument.dart';

/// Source of tradable instruments for the Boleta Inteligente.
///
/// The MVP data layer backs this with a hard-coded seed list (Nasdaq
/// futures, XAU/USD), but the domain layer only knows this contract, so
/// swapping in a remote/broker-fed catalog later is a data-layer-only
/// change.
abstract class InstrumentRepository {
  Future<Either<Failure, List<Instrument>>> getAvailableInstruments();

  Future<Either<Failure, Instrument>> getInstrumentBySymbol(String symbol);
}

File: smarttrade-web/src/features/boleta-inteligente/domain/repositories/instrumentRepository.ts
import type { Result } from '@/core/result/result'
import type { Instrument } from '../entities/instrument'

/**
 * Source of tradable instruments for the Boleta Inteligente.
 *
 * The MVP data layer backs this with a hard-coded seed list (Nasdaq
 * futures, XAU/USD), but the domain layer only knows this contract, so
 * swapping in a remote/broker-fed catalog later is a data-layer-only
 * change.
 */
export interface InstrumentRepository {
  getAvailableInstruments(): Promise<Result<Instrument[]>>
  getInstrumentBySymbol(symbol: string): Promise<Result<Instrument>>
}
